nova_vm/src/ecmascript/types/language/number.rs

```diff
@@ -287,6 +287,22 @@ impl Number {
         }
     }
 
+    pub fn is_pos_one(self, agent: &Agent) -> bool {
+        match self {
+            Number::Number(n) => 1.0 == agent[n],
```

Comment:
**note:** Only `Number::Integer` should ever return positive to this. If any other others return true, that's a bug in Nova as it means that our variants are no longer "sound".

Reply:
Should I change this to `unreachable!()` or keep as is?

Reply:
unreachable sounds fine.

Reply:
Looks like our number variants are not "sound" everywhere. I reverted changing it to `unreachable!()` with an added note.
```
	"built-ins/Math/log10/Log10-specialVals.js" -- Expected Pass, got Crash
	"built-ins/Math/log/S15.8.2.10_A3.js" -- Expected Pass, got Crash
	"built-ins/Math/log2/log2-basicTests.js" -- Expected Pass, got Crash
	"built-ins/Math/acosh/nan-returns.js" -- Expected Pass, got Crash
	"built-ins/Math/log/S15.8.2.10_A2.js" -- Expected Pass, got Crash
```